File: assignment1.cpp
#include <iostream>
#include <ctime>
using namespace std;

void swap(int &a, int &b); // 위치교환
void Sort(int* a, int n); // 정렬함수
static int gcd(int x, int y); // 두 정수의 최대 공약수
static int gcdArray(int* a, int n); // 배열 내 최대 공약수
int getChe(int a, int b); // 소수함수

int main()
{   
    clock_t start, finish; // 시간측정
    double duration; // 시간
    
    int num = 30; // 배열 인덱스 수 초기화
    int arr[num]; // 입력값 배열

    // 배열 인덱스 수 입력
    while (true)
    {   
        cout << "Input the number of numbers to process : ";
        cin >> num;
        try
        {
            if(!cin)
                throw num;
            else if(num <= 1 || num > 30)
                throw num;
            else
                break;
        }
        catch(int num)
        {
            cout << "Input the number over 0 or only Positive Interger! " << endl;
            cin.clear();
            cin.ignore(INT_MAX, '\n');
        }
        
        
    }
    
    // 배열 원소 입력받기
    // 입력오류 예외처리
    cout << "Input the numbers to be processed :  ";
    start = clock();
    for(int i = 0; i<num; i++)
    {
        cin >> arr[i];
    }

    // 배열내 최대공약수 구하기
    cout << endl;
    cout << "GCD of input numbers is " << gcdArray(arr, num) << endl;
    cout << endl;

    // 배열 정렬
    Sort(arr, num);

    // 중복제거하기
    int nArr[num];
    for(int i = 0; i < num; i++)
    {
        nArr[i] = 0;
    }
    int count = 0;
    for(int i = 0; i < num; i++)
    {
        if(arr[i] != arr[i+1])
            nArr[count++] = arr[i];
        
    }

    // 두 수 사이의 Prime number 구하기
    for(int i = 0; i<count-1; i++)
    {
        cout << "Number of prime numbers between " << nArr[i] << 
        ", " << nArr[i+1] << " : " << getChe(nArr[i], nArr[i+1]) << endl;
    }

    finish = clock();

    duration = (double)(finish-start) / CLOCKS_PER_SEC;
    cout << endl;
    cout << "Total execution time using C++ is " << duration << " seconds!" << endl;



    return 0;

}

void swap(int &a, int &b)
{
    int temp = a;
    a = b;
    b = temp;
}

void Sort(int* a, int n)
{
    for(int i=0; i<n; i++)
    {
        int minn = a[i];
        int location = i;
        for(int j = i+1; j<n; j++)
        {
            if(minn>a[j])
            {
                minn = a[j];
                location = j;
            }
        }
        swap(a[i],a[location]);
    }
}


static int gcd(int a, int b)
{
    if(a == 0)
        return b;
    else
        return gcd(b%a, a);
}

static int gcdArray(int* a, int n)
{   
    int result = a[0];
    for(int i = 1; i<n; i++)
    {
        result = gcd(a[i], result);

        if(result == 1)
        {
            return 1;
        }
    }
    return result;
}

// 두 정수사이의 소수갯수
int getChe(int a, int b)
{   
    int flag, i;
    int count = 0;
    while( a < b)
    {
        flag = 0;

        if(a<= 1)
        {
            ++a;
            continue;
        }

        for(i = 2; i <= a/2; ++i)
        {
            if(a%i == 0)
            {
                flag = 1;
                break;
            }
        }

        if(flag == 0)
            ++count;
            ++a;
    }

    return count;
}

File: assignment1.py
import time
import math

def sort(lst, n):
    for i in range(0,n):
        min = i
        for j in range(i+1, n):
            if lst[j] < lst[min]:
                min = j
        lst[i], lst[min] = lst[min], lst[i]

def gcd(a, b):
    if b == 0:
        return a
    else:
        return gcd(b, a % b)

def gcdarray(arr):
    gcd_arr = None
    for i in range(len(arr)):
        if i == 0:
            gcd_arr = arr[i]
        else:
            gcd_arr = gcd(gcd_arr, arr[i])
    return gcd_arr

def getChe(a, b):
    flag = 0
    count = 0
    while(a<b):
        flag = 0
        if(a <= 1):
            a += 1
            continue
        for i in range (2, int(a/2)):
            if(a%i == 0):
                flag = 1
                break
        if(flag == 0):
            count += 1
        a += 1
    return count

num = 30
arr = []
err = 0

while(True):
    try:
        num = int(input("Input the number of numbers to process : "))
    except ValueError:
        print("Input number between 2 and 30")
    if num not in range(2, 30):
        print("Input integer between 2 and 30")
    else:
        break

while(True):
    err = 0
    arr = list(map(int, input("Input the numbers to be processed : ").split()))
    for i in range(0, num):
        if arr[i] > 100000 or arr[i] < 1:
            err = 1
    if err > 0:
        print("Input number between 1 and 100000")
    else:
        break

start = time.time()
math.factorial(100000)
end = time.time()
print(f"GCD of input number is : {gcdarray(arr)}")

sort(arr, num)
nArr = []
count = 0

for i in range(0, len(arr)):
    if i == 0:
        nArr.append(arr[i])
    elif arr[i] != arr[i-1]:
        nArr.append(arr[i])

for i in range(1, len(nArr)):
    print("Number of prime number between", nArr[i-1], ", ", nArr[i], " : ", getChe(nArr[i-1], nArr[i]))

print(f"Total execution time using python is {end - start:.5f} seconds")